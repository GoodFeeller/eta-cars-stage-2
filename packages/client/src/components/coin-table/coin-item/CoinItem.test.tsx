import { render, screen } from '@testing-library/react'
import CoinItem from './CoinItem'
import { ICoin } from '../../../types/ICoin'
import { BrowserRouter } from 'react-router-dom'
import userEvent from '@testing-library/user-event'
import { FunctionComponent } from 'react'

const coin: ICoin = {
  rank: '1',
  symbol: 'BTC',
  name: 'Bitcoin',
  id: 'bitcoin',
  priceUsd: '21000.2313125256326',
  maxSupply: '1000000.223',
  supply: '141312',
  marketCapUsd: '32131412123.231242412321',
  changePercent24Hr: '0.24',
}
const coinBlack: ICoin = {
  rank: '1',
  symbol: 'BTC',
  name: 'Bitcoin',
  id: 'bitcoin',
  priceUsd: '21000.2313125256326',
  maxSupply: '1000000.223',
  supply: '141312',
  marketCapUsd: '32131412123.231242412321',
  changePercent24Hr: '0',
}
const coinRed: ICoin = {
  rank: '1',
  symbol: 'BTC',
  name: 'Bitcoin',
  id: 'bitcoin',
  priceUsd: '21000.2313125256326',
  maxSupply: '1000000.223',
  supply: '141312',
  marketCapUsd: '32131412123.231242412321',
  changePercent24Hr: '-0.24',
}
const emtyCoin: ICoin = {
  rank: '1',
  symbol: 'BTC',
  name: 'Bitcoin',
  id: 'bitcoin',
  priceUsd: '0.0313125256326',
  maxSupply: '1000000.223',
  supply: '141312',
  marketCapUsd: '0.001242412321',
  changePercent24Hr: '0.24',
}
const TestComponent: FunctionComponent = () => {
  return <CoinItem data-testid='item' item={emtyCoin} />
}
describe('CoinItem test', () => {
  test('Default item snapshot', () => {
    render(
      <BrowserRouter>
        <table>
          <tbody>
            <CoinItem item={coin} />
          </tbody>
        </table>
      </BrowserRouter>,
    )
    const tr = screen.queryByTestId('tr')
    expect(tr).toMatchSnapshot()
  })
  test('Empty item test', () => {
    render(
      <BrowserRouter>
        <table>
          <tbody>
            <CoinItem item={emtyCoin} />
          </tbody>
        </table>
      </BrowserRouter>,
    )
    const tr = screen.queryByTestId('tr')
    expect(tr).toBeNull()
  })
  test('Empty item test snapshot', () => {
    render(
      <BrowserRouter>
        <table>
          <tbody>
            <TestComponent />
          </tbody>
        </table>
      </BrowserRouter>,
    )
    const item = screen.queryByTestId('item')
    expect(item).toMatchSnapshot()
  })
  test('Navigate test, must be http://localhost/con/bitcoin', () => {
    render(
      <BrowserRouter>
        <table>
          <tbody>
            <CoinItem item={coin} />
          </tbody>
        </table>
      </BrowserRouter>,
    )
    const tr = screen.queryByTestId('tr') as HTMLElement
    userEvent.click(tr)
    expect(document.location.href).toContain('http://localhost/coin/bitcoin')
  })
  test('Percent must be green', () => {
    render(
      <BrowserRouter>
        <table>
          <tbody>
            <CoinItem item={coin} />
          </tbody>
        </table>
      </BrowserRouter>,
    )
    const percent = screen.queryByTestId('percent') as HTMLElement
    expect(percent).toMatchSnapshot()
  })
  test('Percent must be black', () => {
    render(
      <BrowserRouter>
        <table>
          <tbody>
            <CoinItem item={coinBlack} />
          </tbody>
        </table>
      </BrowserRouter>,
    )
    const percent = screen.queryByTestId('percent') as HTMLElement
    expect(percent).toMatchSnapshot()
  })
  test('Percent must be red', () => {
    render(
      <BrowserRouter>
        <table>
          <tbody>
            <CoinItem item={coinRed} />
          </tbody>
        </table>
      </BrowserRouter>,
    )
    const percent = screen.queryByTestId('percent') as HTMLElement
    expect(percent).toMatchSnapshot()
  })
})
