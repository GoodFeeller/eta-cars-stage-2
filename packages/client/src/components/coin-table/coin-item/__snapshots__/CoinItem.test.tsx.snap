// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinItem test Default item snapshot 1`] = `
<tr
  data-testid="tr"
>
  <td>
    1
  </td>
  <td
    class="imgTd"
  >
    <img
      alt="Not found"
      src="https://assets.coincap.io/assets/icons/btc@2x.png"
    />
    BTC
  </td>
  <td>
    $21,000.23
  </td>
  <td>
    $32,131,412,123.23
  </td>
  <td
    data-testid="percent"
    style="color: green;"
  >
    0.24%
  </td>
  <td
    class="btnTd"
  >
    <button
      class="addButton"
    >
      ADD
    </button>
  </td>
</tr>
`;

exports[`CoinItem test Empty item test snapshot 1`] = `null`;

exports[`CoinItem test Percent must be black 1`] = `
<td
  data-testid="percent"
  style="color: black;"
>
  0%
</td>
`;

exports[`CoinItem test Percent must be green 1`] = `
<td
  data-testid="percent"
  style="color: green;"
>
  0.24%
</td>
`;

exports[`CoinItem test Percent must be red 1`] = `
<td
  data-testid="percent"
  style="color: red;"
>
  -0.24%
</td>
`;
